### PR TITLE
crowbar-testbuild: rewrite in ruby, fully configurable

### PR DESCRIPTION
**Pull request by jdsn**
The redeployment of our CI on the SAP side requires our tools to be more flexible and configurable to allow them to be run in different environments.

Making crowbar-testbuild configurable needed to touch almost every line of the code.
Thus it was a good opportunity for a clean rewrite in ruby.

The new script will be merged first, then a temporary test job will do more test runs, then the old job will be switched to the new script.

### CODE REVIEW

**Comment by dirkmueller** [comment]
I'm not sure I understand the purpose of the rewrite other than splitting stuff into a yaml file (which also is possible in python). can you summarize the requirements or is that documented somewhere? its hard for me to review this besides that it had different implementation choices being made.

**Comment by jdsn** [comment]
@dirkmueller
> (which also is possible in python)

Sure, that would have been also an option. In my initial comment I also addressed this. Going through the code I realized that it would have required to touch almost all code lines. So it would be not been less error prone.

 The requirement is that the tool can be used at SAP. They have a different setup of their internal BuildService, different project names, different parameters for their builds, different NFS server, different mount paths, ... a.s.o. So all this needs to be configurable to make it work on both sides.

Our default configuration (that was previously hardcoded) is now also in this repo and will be used if no other config was found or defined. That way SAP can provide a config that fits their environment to use this tool in their CI.

Other differences/requirements:
- Triggering the gating build has now to be enabled with a parameter. This should allow to run this tool standalone locally and just build a PR package without triggering any job.
- The jenkins job already sets the github status, so it has been removed from this tool entirely. That should also help with running it locally without setting an unwanted status.
- The patch to the package should be fetched via diff (and in a second try via patch) url from github. We already saw builds that only work with diff but failed with patch.

**Comment by aspiers** [comment]
I would have greatly preferred this not to be rewritten in another language.  Incremental changes are much easier to review, and I didn't hear any compelling reason for the rewrite.  Also why was this merged with only one +1?  I was about to suggest some improvements.

**Comment by jdsn** [comment]
@aspiers 
[ 'bmwiedemann', 'rsalevsky' ].size > 1 == true

**Comment by aspiers** [comment]
@jdsn I think you missed my edits.

**Comment by jdsn** [comment]
@aspiers nope, I saw them
